src/cockpit/samples.py

```diff
@@ -301,6 +301,60 @@ def sample(self, samples: Samples) -> None:
                     samples['cgroup.cpu.usage'][cgroup] = usage_nsec / 1000000
 
 
+class CGroupDiskIO(Sampler):
+    descriptions = [
+        SampleDescription('disk.cgroup.read', 'bytes', 'counter', True),
+        SampleDescription('disk.cgroup.written', 'bytes', 'counter', True),
+    ]
+
+    @staticmethod
+    def get_cgroup_name(fd: int) -> Optional[str]:
+        cgroup_name = ""
+        try:
+            with Handle.open('cgroup', os.O_RDONLY, dir_fd=fd) as cgroup_fd:
+                cgroup_name = os.read(cgroup_fd, 1024).decode().strip()
+        except FileNotFoundError:
+            return None
+
+        # Skip leadin ::0/
+        return cgroup_name[4:]
+
+    @staticmethod
+    def get_proc_io(fd: int) -> Optional[List[int]]:
+        data = ""
+        try:
+            with Handle.open('io', os.O_RDONLY, dir_fd=fd) as io_fd:
+                data = os.read(io_fd, 1024).decode()
```

Comment:
For cgroups 1024 made me think "...hopefully this is enough?".

For the "io" file I feel like it's definitely not enough.  This is a very small file, but the format suggests that it might be extended at some point.  Although probably at that point the items we're looking for will still be at the front....

I feel like there's little benefit here to being so conservative with this number.